presto-main/src/main/java/com/facebook/presto/cost/CostCalculatorUsingExchanges.java

```diff
@@ -128,6 +129,22 @@ public PlanNodeCostEstimate visitAssignUniqueId(AssignUniqueId node, Void contex
             return cpuCost(getStats(node).getOutputSizeInBytes(ImmutableList.of(node.getIdColumn()), types));
         }
 
+        @Override
+        public PlanNodeCostEstimate visitRowNumber(RowNumberNode node, Void context)
+        {
+            List<Symbol> symbols = node.getOutputSymbols();
+            if (!node.getMaxRowCountPerPartition().isPresent()) {
+                symbols = ImmutableList.<Symbol>builder()
+                        .addAll(node.getPartitionBy())
+                        .add(node.getRowNumberSymbol())
+                        .build();
+            }
+            PlanNodeStatsEstimate stats = getStats(node);
+            double cpuCost = stats.getOutputSizeInBytes(symbols, types);
```

Comment:
> But when this does not hold, what are the extra symbols we might have on symbols list?

All the other symbols that are not partition keys and not row number. In case `getMaxRowCountPerPartition` is present, the `RowNumberOperator` copies all the symbols.